lineman/app/services/modal_service.coffee

```diff
@@ -0,0 +1,25 @@
+angular.module('loomioApp').factory 'ModalService', ($modal, $rootScope) ->
+  new class ModalService
+    openModal: (name, options = {}) ->
+      return console.log "Invalid modal name: #{name}" unless @modals[name]
+      $rootScope.$broadcast 'modalOpened'
+      $modal.open
+        templateUrl: @modals[name].template
+        controller:  @modals[name].controller
+        resolve:     @setLocals name, options
+
+    setLocals: (name, options) ->
+      locals = {}
+      _.each @modals[name].locals, (local) ->
+        locals[local] = options[local]
+      locals
+
+    modals:
```

Comment:
yay thanks for trying to get on top of this. I think you're on track but I don't think this list of modal config options in here is right. it seems weird that the modal service would need to be added to every time you want a new kind of modal.